src/ngraph_encapsulate_clusters.cc

```diff
@@ -455,6 +458,7 @@ Status EncapsulateClusters(Graph* graph, int graph_id) {
 
     auto node_def =
         NGraphClusterManager::GetClusterGraph(cluster_idx)->add_node();
+    cluster_indices_for_this_rewrite.insert(cluster_idx);
```

Comment:
collecting those indices that are being encapsulated in this rewrite pass only. In general we don't want to use ALL the contents of ClusterManager, since it has leftovers from other independent rewrite passes.

Reply:
Everything we do in the EncapsulateClusters() is for the current rewrite pass alone, no? Dont understand this comment

Reply:
lets f2f

Reply:
Please change the name cluster_indices_for_this_rewrite  to cluster_indices_for_this_graph maybe